--- frmts/hfa/hfadataset.cpp.orig	2012-09-05 11:04:18.676393407 +1200
+++ frmts/hfa/hfadataset.cpp	2012-09-05 11:18:49.204858011 +1200
@@ -1795,13 +1795,15 @@
         }
         else if( poRAT->GetTypeOfCol(col) == GFT_String )
         {
-            unsigned int nMaxNumChars = 0;
+            unsigned int nMaxNumChars = 0, nNumChars;
             /* find the length of the longest string */
             for( int i = 0; i < nRowCount; i++)
             {
-                if(nMaxNumChars < strlen(poRAT->GetValueAsString(i,col)))
+                /* Include terminating byte */
+                nNumChars = strlen(poRAT->GetValueAsString(i,col)) + 1;
+                if(nMaxNumChars < nNumChars)
                 {
-                    nMaxNumChars = strlen(poRAT->GetValueAsString(i,col));
+                    nMaxNumChars = nNumChars;
                 }
             }
        
