--- frmts/hfa/hfafield.cpp.orig	2012-08-10 17:03:33.943178255 +1200
+++ frmts/hfa/hfafield.cpp	2012-08-10 17:05:26.692079668 +1200
@@ -722,6 +722,12 @@
                 HFAStandard( 8, &dfNumber ); 
                 memcpy( pabyData + 12 + nIndexValue * 8, &dfNumber, 8 ); 
             }
+            else if (nBaseItemType == EPT_u8)
+            {
+                // I have no idea what the correct hungarian prefix should be
+                unsigned char nNumber = (unsigned char)dfDoubleValue;
+                memcpy( pabyData + 12 + nIndexValue, &nNumber, 1);
+            }
             else
             {
                 CPLError( CE_Failure, CPLE_AppDefined, 
