--- frmts/hfa/hfaband.cpp.orig	2012-07-22 13:35:05.271809051 +1200
+++ frmts/hfa/hfaband.cpp	2012-07-22 15:25:31.418809401 +1200
@@ -2053,12 +2053,23 @@
     }
 
 /* -------------------------------------------------------------------- */
+/*      Are we compressed? If so, overview should be too                */
+/*      Check RasterDMS like HFAGetBandInfo                             */
+/* -------------------------------------------------------------------- */
+
+    int nCompressionType = 0;
+    HFAEntry *poDMS = poNode->GetNamedChild( "RasterDMS" );
+
+    if( poDMS != NULL )
+        nCompressionType = poDMS->GetIntField( "compressionType" );
+
+/* -------------------------------------------------------------------- */
 /*      Create the layer.                                               */
 /* -------------------------------------------------------------------- */
     osLayerName.Printf( "_ss_%d_", nOverviewLevel );
 
     if( !HFACreateLayer( psRRDInfo, poParent, osLayerName, 
-                         TRUE, 64, FALSE, bCreateLargeRaster, FALSE,
+                         TRUE, 64, nCompressionType, bCreateLargeRaster, FALSE,
                          nOXSize, nOYSize, nOverviewDataType, NULL,
                          nValidFlagsOffset, nDataOffset, 1, 0 ) )
         return -1;
