--- gcore/gdalrasterband.cpp.orig	2012-07-10 16:21:45.094007379 +1200
+++ gcore/gdalrasterband.cpp	2012-07-10 16:36:52.728858490 +1200
@@ -2770,6 +2770,11 @@
     dfScale = nBuckets / (dfMax - dfMin);
     memset( panHistogram, 0, sizeof(int) * nBuckets );
 
+    double dfNoDataValue;
+    int bGotNoDataValue;
+    dfNoDataValue = GetNoDataValue( &bGotNoDataValue );
+    bGotNoDataValue = bGotNoDataValue && !CPLIsNan(dfNoDataValue);
+
     const char* pszPixelType = GetMetadataItem("PIXELTYPE", "IMAGE_STRUCTURE");
     int bSignedByte = (pszPixelType != NULL && EQUAL(pszPixelType, "SIGNEDBYTE"));
 
@@ -2893,6 +2898,9 @@
                     CPLAssert( FALSE );
                 }
 
+                if( bGotNoDataValue && ARE_REAL_EQUAL(dfValue, dfNoDataValue) )
+                    continue;
+
                 nIndex = (int) floor((dfValue - dfMin) * dfScale);
                 
                 if( nIndex < 0 )
@@ -3077,6 +3085,9 @@
                         return CE_Failure;
                     }
                     
+                    if( bGotNoDataValue && ARE_REAL_EQUAL(dfValue, dfNoDataValue) )
+                        continue;
+
                     nIndex = (int) floor((dfValue - dfMin) * dfScale);
 
                     if( nIndex < 0 )
