--- frmts/hfa/hfaopen.cpp.orig	2013-05-08 11:55:07.002828067 +1200
+++ frmts/hfa/hfaopen.cpp	2013-05-08 11:57:09.239418112 +1200
@@ -57,6 +57,7 @@
  "HistogramParameters",  "lBinFunction.numBins",  "STATISTICS_HISTONUMBINS","Eimg_StatisticsParameters830",
  "HistogramParameters",  "dBinFunction.minLimit", "STATISTICS_HISTOMIN",    "Eimg_StatisticsParameters830",
  "HistogramParameters",  "dBinFunction.maxLimit", "STATISTICS_HISTOMAX",    "Eimg_StatisticsParameters830",
+ "HistogramParameters",  "eBinFunction.binFunctionType","STATISTICS_HISTOBINFUNCTION","Eimg_StatisticsParameters830",
  "StatisticsParameters", "lSkipFactorX",          "STATISTICS_SKIPFACTORX", "",
  "StatisticsParameters", "lSkipFactorY",          "STATISTICS_SKIPFACTORY", "",
  "StatisticsParameters", "dExcludedValues",       "STATISTICS_EXCLUDEDVALUES","",
@@ -2803,6 +2804,7 @@
             int nNumBins = poEntry->GetIntField( "BinFunction.numBins" );
             double dMinLimit = poEntry->GetDoubleField( "BinFunction.minLimit" );
             double dMaxLimit = poEntry->GetDoubleField( "BinFunction.maxLimit" );
+            const char *pszBinFunctionType = poEntry->GetStringField( "BinFunction.binFunctionType" );
             
             // fill the descriptor table - check it isn't there already
             poEntry = poNode->GetNamedChild( "Descriptor_Table" );
@@ -2820,7 +2822,7 @@
             poBinFunc->SetIntField( "numBins", nNumBins );
             poBinFunc->SetDoubleField( "minLimit", dMinLimit );
             poBinFunc->SetDoubleField( "maxLimit", dMaxLimit );
-            poBinFunc->SetStringField( "binFunctionType", "linear" ); // we use always a linear
+            poBinFunc->SetStringField( "binFunctionType", pszBinFunctionType );
 
             // we need a child named histogram
             HFAEntry * poHisto = poEntry->GetNamedChild( "Histogram" );
