--- frmts/hfa/hfaband.cpp.orig	2013-05-08 12:10:00.866027535 +1200
+++ frmts/hfa/hfaband.cpp	2013-05-08 12:10:47.152794184 +1200
@@ -2057,6 +2057,16 @@
 	    return -1;
 	}
     }
+/* -------------------------------------------------------------------- */
+/*      Are we compressed? If so, overview should be too                */
+/*      Check RasterDMS like HFAGetBandInfo                             */
+/* -------------------------------------------------------------------- */
+
+    int nCompressionType = 0;
+    HFAEntry *poDMS = poNode->GetNamedChild( "RasterDMS" );
+
+    if( poDMS != NULL )
+        nCompressionType = poDMS->GetIntField( "compressionType" );
 
 /* -------------------------------------------------------------------- */
 /*      Create the layer.                                               */
