--- gcore/gdalrasterband.cpp.orig	2013-05-08 12:14:42.458507544 +1200
+++ gcore/gdalrasterband.cpp	2013-05-08 12:16:03.420338408 +1200
@@ -2736,6 +2736,11 @@
     dfScale = nBuckets / (dfMax - dfMin);
     memset( panHistogram, 0, sizeof(int) * nBuckets );
 
+    double dfNoDataValue;
+    int bGotNoDataValue;
+    dfNoDataValue = GetNoDataValue( &bGotNoDataValue );
+    bGotNoDataValue = bGotNoDataValue && !CPLIsNan(dfNoDataValue);
+
     const char* pszPixelType = GetMetadataItem("PIXELTYPE", "IMAGE_STRUCTURE");
     int bSignedByte = (pszPixelType != NULL && EQUAL(pszPixelType, "SIGNEDBYTE"));
 
@@ -2858,6 +2863,8 @@
                   default:
                     CPLAssert( FALSE );
                 }
+                if( bGotNoDataValue && ARE_REAL_EQUAL(dfValue, dfNoDataValue) )
+                    continue;
 
                 nIndex = (int) floor((dfValue - dfMin) * dfScale);
                 
@@ -3043,6 +3050,9 @@
                         return CE_Failure;
                     }
                     
+                    if( bGotNoDataValue && ARE_REAL_EQUAL(dfValue, dfNoDataValue) )
+                        continue;
+
                     nIndex = (int) floor((dfValue - dfMin) * dfScale);
 
                     if( nIndex < 0 )
